lib/collection/src/collection/resharding.rs

```diff
@@ -214,8 +214,49 @@ impl Collection {
             },
         }
 
+        let is_in_progress = match self.resharding_state().await {
+            Some(state) if state.matches(&resharding_key) => {
+                if !force && state.stage >= ReshardStage::ReadHashRingCommitted {
+                    return Err(CollectionError::bad_request(format!(
+                        "can't abort resharding {resharding_key}, \
+                         because read hash ring has been committed already, \
+                         resharding must be completed",
+                    )));
+                }
+
+                true
+            }
+
+            Some(state) => {
+                log::warn!(
+                    "aborting resharding {resharding_key}, \
+                     but another resharding is in progress:\n\
+                     {state:#?}"
+                );
+
+                false
+            }
+
+            None => {
+                log::warn!(
+                    "aborting resharding {resharding_key}, \
+                     but resharding is not in progress"
+                );
+
+                false
+            }
+        };
```

Comment:
This is already covered by `check_abort_resharding` above. It may not be exactly the same, but I don't see a significant thing that adds value.